lewis_emulators/cryogenic_sms/interfaces/stream_interface.py

```diff
@@ -0,0 +1,238 @@
+
+    def read_direction(self):
+        dir_dict = {"+": "POSITIVE", "-": "NEGATIVE", "0": "ZERO"}
+        return "CURRENT DIRECTION: {}\r\n".format(dir_dict[self._device.direction])
+
+    def write_direction(self, direction):
+        self._device.direction = direction
+        return self._out_message("")
+
+    def read_output_mode(self):
+        return self._out_message("UNITS: {}".format(self._get_output_mode_string()))
+
+    def read_output(self):
+        return "{} OUTPUT: {} {} AT {} VOLTS \r\n".format(self._timestamp(),
+                                                          self._device.output,
+                                                          self._get_output_mode_string(),
+                                                          self._device.heater_value)
+
+    def write_output_mode(self, output_mode):
+        # Convert values if output mode is changing between amps(OFF) / tesla(ON)
+        constant = self._device.constant
+        if output_mode in ["ON", "1"]:
+            if not self._device.is_output_mode_tesla:
+                self._device.switch_mode("TESLA")
+                self._create_log_message("UNITS", "TESLA")
+        elif output_mode in ["OFF", "0"]:
+            if constant == 0:
+                self._device.error_message = "------> No field constant has been entered"
+            else:
+                if self._device.is_output_mode_tesla:
+                    self._device.switch_mode("AMPS")
+                    self._create_log_message("UNITS", "AMPS")
+        else:
+            raise ValueError("Invalid arguments sent")
+        return self._device.log_message
+
+    def read_ramp_target(self):
+        return self._out_message(" RAMP TARGET: {}".format(self._device.ramp_target.name))
+
+    def read_ramp_status(self):
+        output = self._device.output
+        status_message = " RAMP STATUS: "
+        if self._device.is_paused:
+            status_message += "HOLDING ON PAUSE AT {} {}".format(output, self._get_output_mode_string())
+        elif self._device.at_target:
+            status_message += "HOLDING ON TARGET AT {} {}".format(output, self._get_output_mode_string())
+        elif self._device.is_quenched:
+            status_message += "QUENCH TRIP AT {} {}".format(output, self._get_output_mode_string())
+        elif self._device.is_xtripped:
+            status_message += "EXTERNAL TRIP AT {} {}".format(output, self._get_output_mode_string())
+        elif not self._device.at_target and not self._device.is_paused:
+            status_message += "RAMPING FROM {} TO {} {} AT {:07.5f} A/SEC".format(self._device.prev_target,
+                                                                                  self._device.mid_target,
+                                                                                  self._get_output_mode_string(),
+                                                                                  self._device.ramp_rate)

```

Comment:
Presumably the state of the device should be matching exactly one of these clauses. However it's not immediately obvious that that's true.

Maybe we could add a check at the end which will let us know (loudly) if we fail to match any of these conditions?

```suggestion
                                                                                  self._device.ramp_rate)
        else:
            raise ValueError("Didn't match any of the expected conditions")
```